### ISSUE TITLE
Emacs is trying to get non-existing package versions from melpa, and is failing to connect to melp.

### ISSUE THREAD

**Issue by OZoneGuy** [syl20bnr/spacemacs#15659]
#### Description :octocat:
Fail to update or install melpa packages; package files don't exists. Not sure 

#### Reproduction guide :beetle:
- Start Emacs
- Run `Update packages...`, `spc f e U`.

*Observed behaviour:* :eyes: :broken_heart:
Packages fail to install with this error:
```
An error occurred while installing flycheck (error: (file-error https://melpa.org/packages/flycheck-20220612.1800.tar Not found))
```

Curling this URL also returns a 404, and it says it failed to connect to melpa on installation.

*Expected behaviour:* :heart: :smile:
Should install the packages and try to get a valid package link.

Arch Linux. Package information:
```
Name            : emacs
Version         : 28.1-7
Architecture    : x86_64
Build Date      : Wed 27 Apr 2022 06:49:39 PM EDT
Install Date    : Tue 10 May 2022 08:09:52 PM EDT
```

#### Backtrace :paw_prints:
No backtrace for this issue. Just says helm is not installed when I try to use it.

**Comment by lebensterben** [comment]
you need to follow the template and add attach information from SPC h I

**Comment by OZoneGuy** [comment]
I can't use `spc h I`, helm is broken and needs installation, and I copied the template linked in the message when creating the issue.

**Comment by lebensterben** [comment]
Is this a fresh install?
What's the commit hash?

**Comment by OZoneGuy** [comment]
I was trying to update the package after pulling the latest develop branch. So, not a fresh install and on latest develop, as of this comment.

I then tried a fresh install, backed up and moved the `.spacemacs` file, and I am getting a different error...

This is the debug backtrace.
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "bind-map")
  require(bind-map)
  spacemacs-bootstrap/init-bind-map()
  funcall(spacemacs-bootstrap/init-bind-map)
  (let* ((pkg-name (eieio-oref pkg ':name)) (owner (car (eieio-oref pkg ':owners)))) (spacemacs-buffer/message (format "%S -> init (%S)..." pkg-name owner)) (funcall (intern (format "%S/init-%S" owner pkg-name))))
  configuration-layer//configure-package(#<cfgl-package cfgl-package-157fec608bf2>)
  mapc(configuration-layer//configure-package (#<cfgl-package cfgl-package-157fec608bcc> #<cfgl-package cfgl-package-157fec608c16> #<cfgl-package cfgl-package-157fec608bf2> #<cfgl-package cfgl-package-157fec608c2a> #<cfgl-package cfgl-package-157fec608c4e> #<cfgl-package cfgl-package-157fec608c62> #<cfgl-package cfgl-package-157fec608c86> #<cfgl-package cfgl-package-157fec6087a8>))
  (let (packages-to-configure) (let ((--dolist-tail-- packages)) (while --dolist-tail-- (let ((pkg-name (car --dolist-tail--))) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg ...) (spacemacs-buffer/message ...)) ((and ... ...) (spacemacs-buffer/message ...)) ((null ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) (t (let ... ...) (if ... nil ...) (cond ... ...)))) (setq --dolist-tail-- (cdr --dolist-tail--))))) (setq packages-to-configure (reverse packages-to-configure)) (mapc 'configuration-layer//configure-package packages-to-configure) (mapc 'configuration-layer//post-configure-package packages-to-configure))
  configuration-layer//configure-packages-2((async bind-key bind-map diminish evil hydra use-package which-key))
  configuration-layer//configure-packages((abbrev ace-jump-helm-line ace-link ace-window aggressive-indent all-the-icons ansi-colors archive-mode async auto-compile auto-highlight-symbol avy bind-key bind-map bookmark buffer-menu centered-cursor-mode clean-aindent-mode column-enforce-mode compile conf-mode cus-edit dash debug default-helm-config default-ivy-config default-org-config define-word desktop devdocs diminish dired dired-quick-sort dired-x display-fill-column-indicator display-line-numbers doc-view doom-modeline dotenv-mode drag-stuff dumb-jump edebug ediff editorconfig eldoc electric-indent-mode elisp-def elisp-slime-nav emacs-lisp emr ...))
  configuration-layer//load()
  (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configu..." "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading p...")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
  configuration-layer/load()
  (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
  eval-buffer(#<buffer  *load*> nil "/home/omar/.emacs.d/init.el" nil t)  ; Reading at buffer position 3147
  load-with-code-conversion("/home/omar/.emacs.d/init.el" "/home/omar/.emacs.d/init.el" t t)
  load("/home/omar/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x9259cfbc68273aa>) #f(compiled-function () #<bytecode -0x1f3c686ddc0cdc35>) t)
  command-line()
  normal-top-level()
```

**Comment by lebensterben** [comment]
By fresh install, do you mean you have removed ~/.emacs.d?

**Comment by OZoneGuy** [comment]
Yes.

**Comment by lebensterben** [comment]
Not able to reproduce the bug...

You may try to checkout to an earlier commit.

**Comment by OZoneGuy** [comment]
I tried an earlier commit https://github.com/syl20bnr/spacemacs/commit/ba33309327e278a5fe7c12e095ed3a314523484f, but still getting similar error:
```
Package request is unavailable. Is the package name misspelled?
--> installing package: rainbow-delimiters@spacemacs-editing-visual... [33/123]
Package rainbow-delimiters is unavailable. Is the package name misspelled?
--> installing package: quickrun@spacemacs-defaults... [34/123]
Package quickrun is unavailable. Is the package name misspelled?
--> installing package: projectile@spacemacs-project... [35/123]
Package projectile is unavailable. Is the package name misspelled?
--> installing package: popwin@spacemacs-visual... [36/123]
Package popwin is unavailable. Is the package name misspelled?
--> installing package: popup@spacemacs-visual... [37/123]
Package popup is unavailable. Is the package name misspelled?
--> installing package: persp-mode@spacemacs-layouts... [38/123]
Package persp-mode is unavailable. Is the package name misspelled?
--> installing package: password-generator@spacemacs-editing... [39/123]
Package password-generator is unavailable. Is the package name misspelled?
--> installing package: paradox@spacemacs-navigation... [40/123]
Package paradox is unavailable. Is the package name misspelled?
--> installing package: overseer@emacs-lisp... [41/123]
Package overseer is unavailable. Is the package name misspelled?
--> installing package: org-superstar@spacemacs-org... [42/123]
Package org-superstar is unavailable. Is the package name misspelled?
--> installing package: open-junk-file@spacemacs-navigation... [43/123]
Package open-junk-file is unavailable. Is the package name misspelled?
--> installing package: nameless@emacs-lisp... [44/123]
Package nameless is unavailable. Is the package name misspelled?
--> installing package: multi-line@spacemacs-editing... [45/123]
Package multi-line is unavailable. Is the package name misspelled?
--> installing package: macrostep@emacs-lisp... [46/123]
Package macrostep is unavailable. Is the package name misspelled?
--> installing package: lorem-ipsum@spacemacs-editing... [47/123]
Package lorem-ipsum is unavailable. Is the package name misspelled?
--> installing package: link-hint@spacemacs-editing... [48/123]
Package link-hint is unavailable. Is the package name misspelled?
--> installing package: inspector@emacs-lisp... [49/123]
```
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "bind-map")
  require(bind-map)
  spacemacs-bootstrap/init-bind-map()
  funcall(spacemacs-bootstrap/init-bind-map)
  (let* ((pkg-name (eieio-oref pkg ':name)) (owner (car (eieio-oref pkg ':owners)))) (spacemacs-buffer/message (format "%S -> init (%S)..." pkg-name owner)) (funcall (intern (format "%S/init-%S" owner pkg-name))))
  configuration-layer//configure-package(#<cfgl-package cfgl-package-1584cbbf0a08>)
  mapc(configuration-layer//configure-package (#<cfgl-package cfgl-package-1584cbbf09e0> #<cfgl-package cfgl-package-1584cbbf0540> #<cfgl-package cfgl-package-1584cbbf0a08> #<cfgl-package cfgl-package-1584cbbf0564> #<cfgl-package cfgl-package-1584cbbf0578> #<cfgl-package cfgl-package-1584cbbf059c> #<cfgl-package cfgl-package-1584cbbf05b0> #<cfgl-package cfgl-package-1584cbbf05d4>))
  (let (packages-to-configure) (let ((--dolist-tail-- packages)) (while --dolist-tail-- (let ((pkg-name (car --dolist-tail--))) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg ...) (spacemacs-buffer/message ...)) ((and ... ...) (spacemacs-buffer/message ...)) ((null ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) (t (let ... ...) (if ... nil ...) (cond ... ...)))) (setq --dolist-tail-- (cdr --dolist-tail--))))) (setq packages-to-configure (reverse packages-to-configure)) (mapc 'configuration-layer//configure-package packages-to-configure) (mapc 'configuration-layer//post-configure-package packages-to-configure))
  configuration-layer//configure-packages-2((async bind-key bind-map diminish evil hydra use-package which-key))
  configuration-layer//configure-packages((abbrev ace-jump-helm-line ace-link ace-window aggressive-indent all-the-icons ansi-colors archive-mode async auto-compile auto-highlight-symbol avy bind-key bind-map bookmark buffer-menu centered-cursor-mode clean-aindent-mode column-enforce-mode compile conf-mode cus-edit dash debug default-helm-config default-ivy-config default-org-config define-word desktop devdocs diminish dired dired-quick-sort dired-x display-fill-column-indicator display-line-numbers doc-view doom-modeline dotenv-mode drag-stuff dumb-jump edebug ediff editorconfig eldoc electric-indent-mode elisp-def elisp-slime-nav emacs-lisp emr ...))
  configuration-layer//load()
  (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configu..." "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading p...")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
  configuration-layer/load()
  (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
  eval-buffer(#<buffer  *load*> nil "/home/omar/.emacs.d/init.el" nil t)  ; Reading at buffer position 3147
  load-with-code-conversion("/home/omar/.emacs.d/init.el" "/home/omar/.emacs.d/init.el" t t)
  load("/home/omar/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0xf1f8439f2573faa>) #f(compiled-function () #<bytecode -0x1f3c686ddc0cdc35>) t)
  command-line()
  normal-top-level()
```
It looks like a connection issue to the melpa repository. I can curl it without an issue, but I am not sure what is spacemacs doing here.

The warning buffer doesn't exist and there are no warnings in the messages buffer.

Is it possible that it is trying to retrieve versions that don't exist?

**Comment by lebensterben** [comment]
it's more likely you moved to China or a country with bad internet connection...

those packages should be good.

**Comment by OZoneGuy** [comment]
The facts that I am in NA, using no VPN, and have a pretty reliable internet connection makes me think that something is wrong with the program and the configuration and not with my location or internet.

Since I removed all configuration, it suggests the issue is with spacemacs.

But don't worry, looks like it is solved. All I needed to do is run `package-refresh-contents`. Everything is installing like it should now. Wondering why I needed to run that. Never had to do it before.
